sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ConditionEvaluatesToConstant.CSharp6.cs

```diff
@@ -2562,6 +2562,61 @@ public string Method8(string message)
 
             return String.Empty;
         }
+
+        void NullCoalesce_Useless(string a, string b, string c, string d)
+        {
+            string isNull = null;
+            string notNull = "";
+            string notEmpty = "value";
+            string ret;
+
+            ret = b ?? a;
+            ret = b ?? notNull;
+            ret = c ?? notEmpty;
+            ret = d ?? "N/A";
+
+            //Left operand: Values notNull and notEmpty are known to be not-null
+            ret = notNull ?? a;      // Noncompliant
+            ret = ((notNull)) ?? a;      // Noncompliant
+            ret = "Lorem " + (notNull ?? a) + " ipsum"; // Noncompliant
+            ret = notNull ?? "N/A";  // Noncompliant
+            ret = notEmpty ?? "N/A"; // Noncompliant {{Change this expression which always evaluates to 'not null'.}}
+//                ^^^^^^^^
+
+            //Left operand: isNull is known to be null
+            ret = null ?? a;    // Noncompliant
+            ret = isNull ?? a;  // Noncompliant
+            ret = ((isNull)) ?? a;  // Noncompliant
+            ret = "Lorem " + (isNull ?? a) + " ipsum"; // Noncompliant
+
+            //Right operand: isNull is known to be null, therefore ?? is useless
+            ret = a ?? null;    // Noncompliant
+            ret = a ?? isNull;  // Noncompliant {{Change this expression which always evaluates to 'null'.}}
+//                     ^^^^^^
+
+            //Combo/Fatality
+            ret = notNull ?? isNull;    //Noncompliant [LeftA, RightA]
+            ret = isNull ?? null;       //Noncompliant [LeftB, RightB]
```

Comment:
I think these cases are out of scope of the rule. The rule is only about conditions. Although the right hand side is indeed useless, it doesn't really correspond what the rule is supposed to check

Reply:
These two cases are only for documentation purposes to see that it will be raised twice on the same line. Both left and right side are useless in this case.

Reply:
Yes. My point was that the fact that the right hand side is useless is not related to this rule, the lines (2593, 2594)
```
ret = a ?? null;
ret = a ?? isNull;
```
should not raise an issue in my POV, as here, the fact that the right hand side is always null is not related to a condition, which seems to me out of scope of the rule.
Same for right hand side of:
```
ret = notNull ?? isNull;
ret = isNull ?? null;
```

Reply:
So should I drop the check? Or do we have a better *EvaluatesToConstant rule to use?
I agree that this check does not fulfill the root purpose of the rule. It's collateral additional value only.

Reply:
CC @andrei-epure-sonarsource 
Conclusion after discussion with @nicolas-harraudeau-sonarsource is to keep this feature. 
Nicolas pointed out a new POV on this case:
```
ret = a ?? null;
```
the condition `a==null` is **the** useless condition here. The final product will always be the same (value of `a`) no matter value comes in. Therefore it is a "condition" issue and it fits the rule title and description.

I will update RSPEC noncompliant example to make it clear.